src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1176,16 +1176,64 @@ private String authRegistryHeader(final AuthRegistryConfig authRegistryConfig)
     try {
       String authRegistryJson =
         ObjectMapperProvider.objectMapper().writeValueAsString(authRegistryConfig);
+
+      String apiVersion = version().apiVersion();
+      int versionComparison = versionCompare(apiVersion, "1.19");
+
+      // Version below 1.19
+      if (versionComparison < 0) {
+        authRegistryJson = "{\"configs\":" + authRegistryJson + "}";
+      }
+      // Version equal 1.19
+      else if (versionComparison == 0) {
+        authRegistryJson = "{\"auths\":" + authRegistryJson + "}";
+      }
+
       log.debug("Registry Config Json {}", authRegistryJson);
       String authRegistryEncoded = Base64.encodeAsString(authRegistryJson);
       log.debug("Registry Config Encoded {}", authRegistryEncoded);
       return authRegistryEncoded;
-    } catch (JsonProcessingException ex) {
+    } catch (JsonProcessingException | InterruptedException ex) {
       throw new DockerException("Could not encode X-Registry-Config header", ex);
     }
   }
 
   /**
+   * Compares two version strings.
+   * <p>
+   * https://stackoverflow.com/questions/6701948/efficient-way-to-compare-version-strings-in-java
+   * </p>
+   * Use this instead of String.compareTo() for a non-lexicographical
+   * comparison that works for version strings. e.g. "1.10".compareTo("1.6").
+   *
+   * @param str1 a string of ordinal numbers separated by decimal points.
+   * @param str2 a string of ordinal numbers separated by decimal points.
+   * @return The result is a negative integer if str1 is _numerically_ less than str2.
+   * The result is a positive integer if str1 is _numerically_ greater than str2.
+   * The result is zero if the strings are _numerically_ equal.
+   * N.B. It does not work if "1.10" is supposed to be equal to "1.10.0".
+   */
+  private int versionCompare(String str1, String str2) {
```

Comment:
We already have a `versionCompare()` in `DefaultDockerClientTest`. Want to just use that one?